backend/internal/db/dbgorm/deployments.go

```diff
@@ -43,22 +43,24 @@ func (db *gormdb) getDeployment(ctx context.Context, deploymentID string) (*sche
 func (db *gormdb) GetDeployment(ctx context.Context, id sdktypes.DeploymentID) (sdktypes.Deployment, error) {
 	d, err := db.getDeployment(ctx, id.String())
 	if d == nil || err != nil {
-		return nil, err
+		return nil, translateError(err)
 	}
 	return scheme.ParseDeployment(*d)
 }
 
-func (db *gormdb) DeleteDeployment(ctx context.Context, id sdktypes.DeploymentID) error {
-	var b scheme.Deployment
-	if err := db.db.WithContext(ctx).Where("deployment_id = ?", id.String()).Delete(&b).Error; err != nil {
-		return translateError(err)
-	}
-	// FIXME: delete deployment sessions as well?
+func (db *gormdb) deleteDeployment(ctx context.Context, deploymentID string) error {
+	// TODO: this won't work. Consider removing generic delete function.
+	// return delete(db.db, ctx, scheme.Deployment{}, "deployment_id = ?", deploymentID)
 
-	return nil
+	var b scheme.Deployment = scheme.Deployment{DeploymentID: deploymentID}
+	return db.db.WithContext(ctx).Delete(&b).Error
 }
 
-// FIXME: fix generic in order to avoid all this
+func (db *gormdb) DeleteDeployment(ctx context.Context, deploymentID sdktypes.DeploymentID) error {
+	return translateError(db.deleteDeployment(ctx, deploymentID.String()))
+}
+
+// TODO: rewrite generic in order to avoid splitting to Deployment and DeploymentWithStats
```

Comment:
I'm waiting for Itays new generic implementation. If this is something that would allow me to solve this, then good. Otherwise I may consider either remove TODO or add a ticket